core/safeclient/client.go

```diff
@@ -187,7 +200,7 @@ func (c *SafeEthClient) SubscribeFilterLogs(ctx context.Context, q ethereum.Filt
 			rangeStartBlock = 0
 		}
 
-		fromBlock := max(lastBlock, rangeStartBlock) + 1
+		fromBlock := max(lastBlock, rangeStartBlock+1)
```

Comment:
More or less, this is related to this part of the description:
> It also makes it so our log filtering starts from the last block, since we now can detect duplicate blocks - so if, e.g. there were 2 block emissions on lastBlock and we'd only gotten 1 before filtering logs, we don't lose the second one.